proposals/0008-package-manager-testing.md

```diff
@@ -0,0 +1,194 @@
+# Swift Testing
+
+* Authors:
+  [Max Howell](https://github.com/mxcl),
+  [Daniel Dunbar](https://github.com/ddunbar),
+  [Mattt Thompson](https://github.com/mattt)
+* Status: **Review**
+* Review Manager: Rick Ballard
+
+## Introduction
+
+Testing is an essential part of modern software development.
+Tight integration of testing into the Swift Package Manager
+will help ensure a stable and reliable packaging ecosystem.
+
+## Proposed Solution
+
+We propose to extend our conventional package directory layout
+to accomodate test modules.
+Any subdirectory of the package root directory named "Tests"
+or any subdirectory of an existing module directory named "Tests"
+will comprise a test module.
+For example:
+
+    Package
+    ├── Sources
+    │   └── Foo
+    │       └──Foo.swift
+    └── Tests
+        └── Foo
+            └── Test.swift
+
+Or:
+
+    Package
+    └── Sources
+        ├── Foo.swift
+        └── Tests
+            └── Test.swift
+
```

Comment:
Is there a reason for this variant? Will one format be recommended over the other? Having one recommended structure seems like it would be better in terms of consistency/providing guidance to developers.